gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -595,7 +596,12 @@ public GsonBuilder disableHtmlEscaping() {
    */
   @CanIgnoreReturnValue
   public GsonBuilder setDateFormat(String pattern) {
-    // TODO(Joel): Make this fail fast if it is an invalid date format
+    try {
```

Comment:
It appears `null` might actually be a valid `pattern` value to clear any custom pattern previously set for the builder. The changes here would not support that anymore.

@eamonnmcmanus, should `null` be supported?

Apparently no unit test is covering that yet, might be good to add one (or I can add one in a separate PR if desired).

Reply:
if pattern is null, here will  throw null point exception.